resource_library.py

```diff
@@ -33,11 +33,17 @@
 ${resource_src_name} is mapped from src_name by replacing '/' or '.' to '_'.
 
 """
-
+from __future__ import print_function
 import os
 import sys
 
 
+# Assume 'reduce' fuction needs to be imported in this way in future versions.
+version_info = int(sys.version.split('.')[0])
+if version_info > 2:
+    from functools import reduce
```

Comment:
I think we can just use `reduce` from `functools` in python2 and python3 since the doc https://docs.python.org/2/library/functools.html#functools.reduce states that it's the same function with built-in `reduce`

Reply:
Thanks so much for letting me know this, and I prefer to just using "from functools import reduce" directly.

Reply:
So, I will wait until you push a new change